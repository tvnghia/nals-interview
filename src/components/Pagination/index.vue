<template>
  <ul class="pagination">
    <li
      v-for="page in pages"
      :key="page.index"
      @click="clickPagination(page.page)"
      :class="['pagination__item', { active: page.active }]"
    >
      <a class="pagination__link">{{ page.page }}</a>
    </li>
  </ul>
</template>

<script>
const PAGE_TOTAL_DUMMY = 5

export default {
  name: 'Pagination',

  props: {
    value: {
      type: Number,
      require: true
    }
  },

  computed: {
    pages() {
      const pages = []
      for (let i = 0; i < PAGE_TOTAL_DUMMY; i++) {
        pages.push({
          active: this.currentPage === i + 1,
          page: i + 1
        })
      }
      return pages
    },

    currentPage() {
      return parseInt(this.$route.query.page)
    }
  },

  methods: {
    clickPagination(page) {
      if (page === this.currentPage) return

      this.$emit('input', page)
    }
  }
}
</script>
