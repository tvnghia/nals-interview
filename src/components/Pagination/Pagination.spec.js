import { shallowMount } from '@vue/test-utils'
import wrapperOps from '@/config-test/wrapper'
import Pagination from '@/components/Pagination'

const clickPagination = jest.spyOn(Pagination.methods, 'clickPagination')

describe('Pagination.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Pagination, wrapperOps)
  })

  afterEach(() => {
    wrapper.vm.$destroy()
  })

  describe('when Pagination was mounted', () => {
    it('should Pagination render corectly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when switch page', () => {
    it('should call clickPagination method', async () => {
      wrapper.find('li').trigger('click')
      await wrapper.vm.$nextTick()

      expect(clickPagination).toHaveBeenCalled()
      expect(wrapper.emitted('input')).toHaveLength(1)
      expect(wrapper.emitted('input')[0]).toEqual([1])
    })
  })
})
