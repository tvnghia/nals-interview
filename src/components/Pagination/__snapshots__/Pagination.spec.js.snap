// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue when Pagination was mounted should Pagination render corectly 1`] = `
<ul class="pagination">
  <li class="pagination__item"><a class="pagination__link">1</a></li>
  <li class="pagination__item"><a class="pagination__link">2</a></li>
  <li class="pagination__item"><a class="pagination__link">3</a></li>
  <li class="pagination__item"><a class="pagination__link">4</a></li>
  <li class="pagination__item"><a class="pagination__link">5</a></li>
</ul>
`;
